### PR TITLE
Fix trashbin deleteallhooks solution2 (with bulk hook)

### PR DESCRIPTION
**Pull request by stevenbuehner**
see #25942

### CODE REVIEW

File: apps/files_trashbin/lib/Trashbin.php [hunk 1]
```diff
@@ -469,16 +469,59 @@ private static function restoreVersions(View $view, $file, $filename, $uniqueFil
...
 		$user = User::getUser();
 		$view = new View('/' . $user);
+		$fileInfos = $view->getDirectoryContent('files_trashbin/files');
+
+		// Array to store the relative path in (after the file is deleted, the view won't be able to relativise the path anymore)
+		$filePaths = array();
+		foreach($fileInfos as $fileInfo){
+			$filePaths[] = $view->getRelativePath($fileInfo->getPath());
+		}
+		unset($fileInfos); // save memory
```
Comment (jvillafanez) [inline apps/files_trashbin/lib/Trashbin.php]: ❤️

**Comment by jvillafanez** [comment]
Code looks good 👍

**Comment by PVince81** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
squash on merge would have been nice

**Pull request merged**